### PR TITLE
fix(dataseriesformsettings): Set empty strings for missing values

### LINKED ISSUE #3680: [DataSeriesFormSettings] null and undefined values do not reset TextInput values
### What package is this for?

- [X] React
- [ ] Angular

### Describe the bug

TextInput values don't get cleared for parameters not present in new configuration.

### To Reproduce

1. Go to DataSeriesFormSettings
2. Set values to the following attributes/props
      **decimalPrecision**: PropTypes.number,
      **xLabel**: PropTypes.string,
      **yLabel**: PropTypes.string,
      **unit**: PropTypes.string
      **maximumDataPointsLabel**: PropTypes.string
3. Render component.
4. Re-render component with only **xLabel**
5. Inputs associated with missing attributes keep previous value.

> Please create a reduced test case in CodeSandbox:
> https://codesandbox.io/s/github/carbon-design-system/carbon-addons-iot-react/tree/next/packages/react/examples/codesandbox

### Expected behavior

It is expected to clear input value when attribute is absent from configuration.

### Environment/versions:

- OS: MacOS
- Browser: chrome
- `carbon-addons-iot-react` version: v2.152.0-next.13

### Specific timeline issues / requests

Do you want this work within a specific time period? Is it related to an
upcoming release?

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### PR DESCRIPTION
**Pull request by vaffonso**
Closes #3680

**Summary**

- Set empty string in case of null or undefined values

**Change List (commits, features, bugs, etc)**

- DataSeriesFormSettings: use nullish coalescing operator to set empty strings

**Acceptance Test (how to verify the PR)**

- Render component with all attributes from content filled
- Re-render component with some of them missing

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Comment by vaffonso** [comment]
@herleraja @EdmondCShaffer could you take a look?

**Comment by JordanWSmith15** [comment]
@davidicus FYI

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx [hunk 1]
```diff
@@ -144,7 +144,7 @@ const DataSeriesFormSettings = ({ cardConfig, onChange, i18n }) => {
               content: { ...cardConfig.content, maximumDataPoints },
             });
           }}
-          value={content?.maximumDataPoints}
+          value={content?.maximumDataPoints ?? ''}
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx:147]: its numeric value. Is there no other way we can fix this issue ?

File: packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx [hunk 2]
```diff
@@ -127,7 +127,7 @@ const DataSeriesFormSettings = ({ cardConfig, onChange, i18n }) => {
               },
             })
           }
-          value={content?.decimalPrecision}
+          value={content?.decimalPrecision ?? ''}
```
Comment (herleraja) [inline packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormSettings.jsx:130]: I think this is also numeric value.  May be good to add type="number"

**Review by herleraja — COMMENTED** [review]
It's fine to have an empty string literal for labels, but for numeric fields, its better be null or undefined.

**Comment by davidicus** [comment]
Doing the steps outlined in this issue on this [story](https://next.carbon-addons-iot-react.com/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-dashboardeditor--with-initial-value), it seems like the fields get updated.

**Review by davidicus — CHANGES_REQUESTED** [review]
Just the comment on the number inputs.

**Comment by vaffonso** [comment]
Hi @davidicus @herleraja 
I was able to reproduce the issue in this [story](https://next.carbon-addons-iot-react.com/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-dashboardeditor--with-initial-value) following these steps:
- Select timeseries card
- Switch to Settings tab
- Note that Unit, Decimal Precision and Maximum Data Points are empty
- Add new card
- Select Time Series Line
- Give it a title Test
- Switch to Settings tab
- Fill Unit, Decimal Precision and Maximum Data Points with value
- Select timeseries card again
- Note that Unit, Decimal Precision and Maximum data points value are not reset

I have commented in the original defect:

> I see 3 possible approaches:
> 1. Provide empty strings for Data Series Form Settings attributes - Monitor UI
> 2. Check null and undefined values and replace with empty strings - PAL lib (solution tested)
> 3. Apply check for null/undefined check in TextInput - Carbon lib

Should we opt for the 1st option?

**Comment by davidicus** [comment]
@vaffonso can we not provide a null value for these fields?

**Comment by vaffonso** [comment]
I pushed a couple changes to this PR. Instead of empty strings, number inputs from DataSeriesFormSettings receive key prop to force a render.
In addition, field **decimal precision** was changed to number.

**Review by davidicus — APPROVED** [review]
LGTM 👍

**Pull request merged**